src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -928,19 +928,91 @@ DeferErrorIfCannotPushdownSubquery(Query *subqueryTree, bool outerMostQueryHasLi
 					  "functions";
 	}
 
-	if (subqueryTree->limitOffset)
+	/*
+	 * Correlated subqueries are effectively functions that are repeatedly called
+	 * for the values of the vars that point to the outer query. We can liberally
+	 * push down SQL features within such a function, as long as co-located join
+	 * checks are applied.
+	 */
+	if (!contain_vars_of_level((Node *) subqueryTree, 1))
 	{
-		preconditionsSatisfied = false;
-		errorDetail = "Offset clause is currently unsupported when a subquery "
-					  "references a column from another query";
-	}
+		if (subqueryTree->limitOffset)
```

Comment:
maybe this is a good time to move the logic inside the if statement to a new function, like `SubqueryRequiresMergeStep` or such?